### PR TITLE
build_config.py: add --dry-run, --llvm=system, fix CHPL_HOME default

### PR DESCRIPTION
**Pull request by awallace-cray**
Minor updates and improvements to util/build_config.py
Noticed while working https://github.com/chapel-lang/chapel/issues/9551

### CODE REVIEW

**Comment by awallace-cray** [comment]
Not urgent.

File: util/build_configs.py [hunk 1]
```diff
@@ -470,6 +475,11 @@ def _format_text(self, text):
         action='store_true',
         help='Verbose output.'
     )
+    parser.add_option(
+        '-n', '--dryrun',
```
Comment (ben-albrecht) [inline util/build_configs.py]: Maybe nitpicky: `-n` seems like an odd 1-character flag for `--dryrun` to me. Is this a common flag for dry run?

Comment (awallace-cray) [reply util/build_configs.py]:
```
$ man make
...
       -n, --just-print, --dry-run, --recon
            Print the commands that would be executed, but do not execute them.
```
----
```
$ git clean --help
...
       -n, --dry-run
           Don't actually remove anything, just show what would be done.
```

How about I change the long name to --dry-run? That's what I meant anyway! 
Thanks.

Comment (ben-albrecht) [reply util/build_configs.py]: Sounds good to me!

**Pull request merged**